libs/langchain-community/src/llms/googlegenerativeai.ts

```diff
@@ -0,0 +1,363 @@
+
+  get _isGenerateContentModel() {
+    if (/^gemini/.test(this.modelName)) {
+      return true;
+    }
+    return false;
+  }
+
+  constructor(fields?: GoogleGenerativeAITextInput) {
+    super(fields ?? {});
+
+    this.modelName =
+      fields?.modelName?.replace(/^models\//, "") ?? this.modelName;
+
+    const isGenerateContentModel = this._isGenerateContentModel;
+
+    this.temperature = fields?.temperature ?? this.temperature;
+    if (this.temperature && (this.temperature < 0 || this.temperature > 1)) {
+      throw new Error("`temperature` must be in the range of [0.0,1.0]");
+    }
+
+    this.maxOutputTokens = fields?.maxOutputTokens ?? this.maxOutputTokens;
+    if (this.maxOutputTokens && this.maxOutputTokens < 0) {
+      throw new Error("`maxOutputTokens` must be a positive integer");
+    }
+
+    this.topP = fields?.topP ?? this.topP;
+    if (this.topP && this.topP < 0) {
+      throw new Error("`topP` must be a positive integer");
+    }
+
+    if (this.topP && this.topP > 1) {
+      throw new Error("`topP` must in the range of [0,1]");
+    }
+
+    this.topK = fields?.topK ?? this.topK;
+    if (this.topK && this.topK < 0) {
+      throw new Error("`topK` must be a positive integer");
+    }
+
+    this.stopSequences = fields?.stopSequences ?? this.stopSequences;
+
+    this.apiKey = fields?.apiKey ?? getEnvironmentVariable("GOOGLE_API_KEY");
+    if (!this.apiKey) {
+      throw new Error(
+        "Please set an API key for Google GenerativeAI " +
+          "in the environmentb variable GOOGLE_API_KEY " +
+          "or in the `apiKey` field of the " +
+          "GoogleGenerativeAI constructor"
+      );
+    }
+
+    this.safetySettings = fields?.safetySettings ?? this.safetySettings;
+    if (this.safetySettings && this.safetySettings.length > 0) {
+      if (isGenerateContentModel) {
+        assertSafetySettings(
+          this.safetySettings as SafetySetting[],
+          HarmCategory,
+          HarmBlockThreshold
+        );
+      } else {
+        assertSafetySettings(
+          this
+            .safetySettings as protos.google.ai.generativelanguage.v1beta2.ISafetySetting[],
+          protos.google.ai.generativelanguage.v1beta2.HarmCategory,
+          protos.google.ai.generativelanguage.v1beta2.SafetySetting
+            .HarmBlockThreshold
+        );
+      }
+    }
+
+    this.client = isGenerateContentModel
+      ? new GenerativeAI(this.apiKey).getGenerativeModel({
+          model: this.modelName,
+          safetySettings: this.safetySettings as SafetySetting[],
+          generationConfig: {
+            candidateCount: 1,
+            stopSequences: this.stopSequences,
+            maxOutputTokens: this.maxOutputTokens,
+            temperature: this.temperature,
+            topP: this.topP,
+            topK: this.topK,
+          },
+        })
+      : new TextServiceClient({
+          authClient: new GoogleAuth().fromAPIKey(this.apiKey),
```

Comment:
Is the use of GoogleAuth only for backwards compatibility?

(If so - can we reconsider it? Removing that would make this module available in browser-based JavaScript.)

Reply:
Yes, for now new SDK only support new models. Let me check if I can get rid of this module for Auth.

Reply:
I think two different entrypoints is actually better since people would otherwise need to install two peer dependencies instead of just one. At the very least, it's not backwards compatible with PaLM.

Reply:
I've opened #3641 for Google Generative AI - will Vertex require a different SDK? Will it be backwards compatible with the existing Vertex class?

If that's the case I think my preferred approach would be to update the existing vertex class.

Reply:
Yeah, unfortunately Vertex will require a different SDK: https://github.com/googleapis/nodejs-vertexai/
And because currently in langchainjs we have bespoke implementation of VertexAI  it would be a little cumbersome to extend base class.

Reply:
Could we stick with the bespoke one or have things significantly changed? 

Concern would be that it'd be breaking for folks currently using it. What would your ideal import path be?

Reply:
IMO I think quickest path forward is to merge and ship #3641, then worry about Vertex afterwards. It sounds like it shouldn't be in the same package due to dependency web vs. Node compatibility issues anyway.